articles/connections/nativesocial/facebook.md

```diff
@@ -0,0 +1,108 @@
+description: Learn how to add login functionality to your native app with Facebook. 
+topics:
+  - authentication
+  - connections
+  - social
+  - facebook
+contentType: how-to
+useCase:
+  - add-login
+  - customize-connections
+  - add-idp
+---
+# Add Facebook Login to Native Apps
+
+You can add functionality to your native application to allow your users to authenticate using Facebook natively, within the application. This does not require redirection via a web browser.
+
+## How it works
+
+The Native Facebook login flow works as follows:
+
+* **Step 1**: The application authenticates a user via the Facebook SDK and acquires an Access Token.
+* **Step 2**: The application uses that Access Token to request a special [Facebook Session Info Access Token](https://developers.facebook.com/docs/facebook-login/access-tokens/session-info-access-token).
+* **Step 3**: The application can then use the Facebook Session Info token to request a user profile from Auth0.
+
+## Prerequisites
+
+Before you configure Native Facebook login for your native app via Auth0, you must:
+
+1. [Set up your application](/connections/social/facebook) with Facebook and as an Auth0 connection
+1. [Use the relevant Facebook SDK in your application](https://developers.facebook.com/docs/apis-and-sdks/)
+1. Create an application with Auth0 (if you have not already) and enable the Sign in with Facebook option in the [Dashboard > (Your Application) > Settings > Advanced Settings > Device Settings](${manage_url}).
+  ![Native Social Login Settings](/media/articles/connections/nativesocial/native-social-login.png)
+1. Follow the implementation details below
+
+## Implementation details
+
+### Setup in your application
+
+As above, the process to acquire a user profile from Auth0 using Native Facebook login is a three step one, from your application's perspective:
+
+* **Step 1**: The application authenticates a user via the Facebook SDK. It will obtain an Access Token from Facebook, and will now need to request the user profile.
+* **Step 2**: The application uses the Access Token to request a [Facebook Session Info Access Token](https://developers.facebook.com/docs/facebook-login/access-tokens/session-info-access-token).
+
+This request will look similar to the following:
+
+```
+GET https://graph.facebook.com/v5.0/oauth/access_token?
+grant_type=fb_attenuate_token&
+client_id=457704041391802&
+fb_exchange_token=<facebook_access_token>
+```
+
+and the response:
+
+```json
+{
+    "access_token": "XAAGgR4b...1lHWNCpqrAhcpoAZDZD",
+    "token_type": "bearer",
+    "expires_in": 5183924
+}
+```
+
+* **Step 3**: The application can then use this token to request a user profile from Auth0 by calling Auth0’s `/oauth/token` endpoint using the [client credentials exchange flow](/flows/concepts/client-credentials), with the `facebook-session-access-token` token type. If all goes well, Auth0 will return a normal response from the exchange, with the addition of the user profile.
```

Comment:
**Step 3**: The application needs to retrieve the user profile from Facebook using the SDK, which will end in a request similar to the following:
```
GET https://graph.facebook.com/v5.0/<facebook user id>?access_token=<facebook access token>&fields=email,name 
```
**Step 4**: The application can then use the session info access token and the facebook user profile to authenticate with Auth0 by calling Auth0’s `/oauth/token` endpoint using the Token Exchange flow with the `facebook-session-access-token` token type. If all goes well, Auth0 will return a normal response from the exchange, with the addition of the user profile.